yarn-project/end-to-end/src/simulators/token_simulator.ts

```diff
@@ -80,15 +80,20 @@ export class TokenSimulator {
     return this.balancesPrivate.get(address.toString()) || 0n;
   }
 
-  public async check() {
-    expect(await this.token.methods.total_supply().simulate()).toEqual(this.totalSupply);
-
-    // Check that all our public matches
+  public async check(wallet: Wallet) {
```

Comment:
Don't love this, works when we have just one wallet, but with multiple as we will need in escrow it gets funky, think we could set a default when we are constructing or something 🤷, But I can deal with that at the escrow instead. 

Think we might also run into some issues if we are moving towards more calls than what a single batch can handle, or will it already be handled in the batch 🤔

Reply:
Haven't seen the escrow yet, was planning to check it out in a different PR to avoid merge hell, but yeah, it won't work with different wallets interacting (this was mainly to get it working and see if the time was somewhat reasonable)

The batching doesn't handle +4 fns per tx, we have to do it externally (we have just been lucky for the time being, but creating a private getter insteat of the current unconstrained will make it blow up). I can do it now if we anticipate the check function to be called with +4 constrained calls.